lib/main_containers.pm

```diff
@@ -95,9 +95,10 @@ sub load_host_tests_podman {
 }
 
 sub load_host_tests_docker {
-    load_container_engine_test('docker');
-    load_image_test('docker');
-    load_3rd_party_image_test('docker');
+    my ($run_args) = @_;
+    load_container_engine_test('docker', $run_args);
```

Comment:
Please take a look how the `load_container_engine_test` subroutine is defined. It passes `run_args` parameter to `loadtest`.

Reply:
`$run_args` is needed for the publiccloud test runs, which are using the same routines as the "normal" test runs.
Before we were creating a new instance of the `$run_args` in each subroutine, which is wrong because the important variables like instance are then not shared between them.

This PR moves the creation of `$run_args` to one place and then passes those variables to each test run, which was creating a new instance before, thus violating the singleton pattern.

Reply:
ok, thanks for clarification.